cpp/src/io/comp/nvcomp_adapter.cpp

```diff
@@ -527,6 +621,40 @@ void batched_decompress(compression_type compression,
 }
 
 // Wrapper for nvcompBatched<format>CompressGetMaxOutputChunkSize
+#if NVCOMP_VER_MAJOR >= 5
+size_t compress_max_output_chunk_size(compression_type compression,
+                                      uint32_t max_uncompressed_chunk_bytes)
+{
+  auto const capped_uncomp_bytes = std::min<size_t>(
+    compress_max_allowed_chunk_size(compression).value_or(max_uncompressed_chunk_bytes),
+    max_uncompressed_chunk_bytes);
+
+  size_t max_comp_chunk_size = 0;
+  nvcompStatus_t status      = nvcompStatus_t::nvcompSuccess;
+  switch (compression) {
+    case compression_type::SNAPPY:
+      status = nvcompBatchedSnappyCompressGetMaxOutputChunkSize(
+        capped_uncomp_bytes, nvcompBatchedSnappyCompressDefaultOpts, &max_comp_chunk_size);
+      break;
+    case compression_type::DEFLATE:
```

Comment:
```suggestion
    case compression_type::DEFLATE: [[fallthrough]];
```
The [fallthrough](https://en.cppreference.com/w/cpp/language/attributes/fallthrough.html) attribute helps tell the compiler and linters that this was intentional.